gnat2goto/driver/tree_walk.adb

```diff
@@ -1289,8 +1308,28 @@ package body Tree_Walk is
                when Attribute_Range  =>
                   return Report_Unhandled_Node_Irep (N, "Do_Expression",
                                                      "Range attribute");
-               when Attribute_First  => return Do_Array_First (N);
-               when Attribute_Last   => return Do_Array_Last (N);
+               when Attribute_First  =>
+                  if Nkind (Etype (Prefix (N))) = N_Defining_Identifier
+                    and then
+                      Get_Name_String (Chars (Etype (Etype (Prefix (N)))))
+                    = "string"
+                  then
+                     return Report_Unhandled_Node_Irep (N, "Do_Expression",
```

Comment:
These errors seem to be triggered in UKNI which, I don't think, includes function access types.  Are they solving a different problem, if so, given they are being triggered, are they solving it correctly?

Reply:
These error were always there, just hidden as raised exceptions. They are unrelated to function pointers, but our supporting of function pointers uncovered them, as exceptions. I turned some exceptions into unsupported features so that's it's easier to investigate them.

Reply:
Brilliant; thank you.